helios-core/src/main/kotlin/helios/instances/instancesArrow.kt

```diff
@@ -29,6 +30,16 @@ fun <A, B> Either.Companion.decoder(decoderA: Decoder<A>, decoderB: Decoder<B>)
       { v -> v.right().map { it.right() } })
 }
 
+fun <A> NonEmptyList.Companion.encoder(encoderA: Encoder<A>) = object : Encoder<NonEmptyList<A>> {
+  override fun NonEmptyList<A>.encode(): Json =
+    ListEncoderInstance(encoderA).run { all.encode() }
+}
+
+fun <A> NonEmptyList.Companion.decoder(decoderA: Decoder<A>) = object : Decoder<NonEmptyList<A>> {
+  override fun decode(value: Json): Either<DecodingError, NonEmptyList<A>> =
+    ListDecoderInstance(decoderA).decode(value).flatMap { NonEmptyList.fromList(it).toEither { ArrayDecodingError(value) } }
```

Comment:
Is it worth a new kind of error?

Reply:
The errors refer to Json types right now they could be more explicit about this indeed